lang/py/src/avro/io.py

```diff
@@ -119,6 +124,9 @@ def validate(expected_schema, datum):
     return (isinstance(datum, int) or isinstance(datum, long)
             or isinstance(datum, float))
   elif schema_type == 'fixed':
+    if hasattr(expected_schema, 'logical_type') and \
+                    expected_schema.logical_type == 'decimal':
```

Comment:
This checks if the datum for the particular schema is of the right type or not. It doesn't check for the actual schema. Example, the datum for decimal cannot be a `list` or an `int`. It has to be a `str`.

Ill remove the backslashes (though that's what's been used in other places)